pinot-common/src/main/java/org/apache/pinot/common/function/scalar/DateTimeFunctions.java

```diff
@@ -642,4 +1138,30 @@ public static long timestampDiff(String unit, long timestamp1, long timestamp2)
     ISOChronology chronology = ISOChronology.getInstanceUTC();
     return DateTimeUtils.getTimestampField(chronology, unit).getDifferenceAsLong(timestamp2, timestamp1);
   }
+  @ScalarFunction(names = {"timestampDiffMVMV", "dateDiffMVMV"})
+  public static long[] timestampDiffMVMV(String unit, long[] timestamp1, long[] timestamp2) {
+    assert timestamp1.length == timestamp2.length;
```

Comment:
Don't use assert here as it won't be executed in production. We should explicitly check and throw exception if the length doesn't match